docs/tuplets.md

```diff
@@ -0,0 +1,123 @@
+## API Design
+### API Definition
+**Open**: Naming is still up for debate. Having trouble with that.
+```swift
+struct Tuplet {
+    /// The notes that make up the tuplet
+    public private(set) var notes: [NoteCollection]
+    /// The number of notes of the specified duration that this tuplet contains
+    public let baseNoteCount: Int
+    /// The duration of the notes that define this tuplet
+    public let baseNoteDuration: NoteDuration
+    /// The number of notes that this tuplet fits in the space of
+    public let noteCountFit: Int
+    
+    init(_ count: Int, _ duration: NoteDuration, inSpaceOf baseCount: Int? = nil, _ baseDuration: NoteDuration? = nil, notes: [NoteCollection]) throws
+    mutating func replaceNote(at index: Int, with note: Note) throws
+}
+```
+### Secondary duration
+We can choose to allow a second duration to be specified like in [Finale](https://usermanuals.finalemusic.com/Finale2014Mac/Content/Finale/TPDLG.htm). However, in order to put the ratio on top, it seems like you need to convert to the same duration?
+
+**Open**: Why have a second duration? Don't we need to convert to the same duration for the ratio?
+
+If we do have a second duration, the initializer would function in the following way:
+- The second duration is optional and will default to the same as the first duration.
+- If the second duration is not the same as the first, we will need to internally convert it into the same as the first duration, so that we can use a standard ratio notation.
+
+### Usage
+If you would like to use a standard ratio, you can specify only the first number and duration. Then, `Tuplet` will validate whether the number specified is a standard (2-9) and if it is, it will automatically set the second number to the static ratio.
+
+*Standard Ratio*
+```swift
+let standardTriplet = try! Tuplet(3, .eighth, notes: [eighthNote, eighthNote, eighthNote])
+```
+*Standard Ratio w/Odd definition*
+```swift
+let standardTriplet = try! Tuplet(3, .eighth, inSpaceOf: 1, .quarter, notes: [eighthNote, eighthNote, eighthNote])
+```
+
+*Custom Ratio*
+```swift
+let customOctuplet = try! Tuplet(
+    8,
+    .eighth,
+    inSpaceOf: 5,
+    .eighth, 
+    notes: [eighthNote, eighthNote, eighthNote, eighthNote, eighthNote, eighthNote, eighthNote, eighthNote]
+)
+```
+
+## Other API Changes
+### NoteCollection
+#### API Before
+```swift
+protocol NoteCollection {
+    var noteCount: Int
+}
+```
+
+#### API After
+**Open**: Naming is still up for debate.
+```swift
+protocol NoteCollection {
+    /**
+     The duration of the note that in combination with `noteTimingCount` 
+     will give you the amount of time this `NoteCollection` occupies.
+     */
+    var noteDuration: NoteDuration
+    /**
+     The number of notes to indicate the amount of time occupied by this
+     `NoteCollection`. Combine this with `noteDuration`.
+     */
+    var noteTimingCount: Int
+    /// The count of actual notes in this `NoteCollection`
+    var noteCount: Int
```

Comment:
What would be the use of noteTimingCount and noteCount? Would it be possible to only use noteDuration? This would imply that noteDuration is no longer an Enum.